examples/machine_translation/train.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import time
+
+import tensorflow as tf
+from absl import app
+from absl import flags
+
+from examples.machine_translation.data import get_dataset_and_tokenizer
+from examples.machine_translation.model import TranslationModel
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_integer("num_epochs", 1, "Number of epochs to train.")
+flags.DEFINE_integer("num_encoders", 2, "Number of Transformer encoder layers.")
+flags.DEFINE_integer("num_decoders", 2, "Number of Transformer decoder layers.")
+flags.DEFINE_integer("batch_size", 64, "The training batch size.")
+flags.DEFINE_float("learning_rate", 0.001, "The initial learning rate.")
+flags.DEFINE_integer("embed_dim", 64, "Embedding size.")
+flags.DEFINE_integer(
+    "intermediate_dim",
+    128,
+    "Intermediate dimension (feedforward network) of transformer.",
+)
+flags.DEFINE_integer(
+    "num_heads",
+    8,
+    "Number of head of the multihead attention.",
+)
+flags.DEFINE_integer(
+    "sequence_length",
+    20,
+    "Input and output sequence length.",
+)
+flags.DEFINE_integer(
+    "vocab_size",
+    15000,
+    "Vocabulary size, required by tokenizer.",
+)
+
+flags.DEFINE_string(
+    "saved_model_path",
+    "saved_models/machine_translation_model",
+    "The path to saved model",
+)
+
+
+def train_loop(model, train_ds, val_ds):
```

Comment:
Is there a reason we can't use `fit` here?

Reply:
Somehow I thought model.fit() does not allow masking, but actually we can use `sample_weights`. 

Changed to model.fit().